From 41ba210f8322970bec42170424c41b801f939114 Mon Sep 17 00:00:00 2001
From: Sebastien Boeuf <sebastien.boeuf@intel.com>
Date: Thu, 19 Feb 2015 11:53:46 +0100
Subject: [PATCH 18/20] gc2155: Fix deadlock on error

As for the ov5648, we detected a possible deadlock in gc2155
driver because the method gc2155_s_config would return without
releasing the mutex.

Signed-off-by: Sebastien Boeuf <sebastien.boeuf@intel.com>
---
 drivers/media/i2c/atomisp/gc2155.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/media/i2c/atomisp/gc2155.c b/drivers/media/i2c/atomisp/gc2155.c
index 98dc1ef..a9432bd 100644
--- a/drivers/media/i2c/atomisp/gc2155.c
+++ b/drivers/media/i2c/atomisp/gc2155.c
@@ -1323,7 +1323,7 @@ static int gc2155_s_config(struct v4l2_subdev *sd,
 		ret = dev->platform_data->platform_init(client);
 		if (ret) {
 			dev_err(&client->dev, "platform init err\n");
-			return ret;
+			goto fail_init_platform;
 		}
 	}
 	ret = power_down(sd);
@@ -1366,6 +1366,7 @@ fail_power_on:
 	power_down(sd);
 	dev_err(&client->dev, "sensor power-gating failed\n");
 fail_power_off:
+fail_init_platform:
 	mutex_unlock(&dev->input_lock);
 	return ret;
 }
-- 
2.1.0

